### PR TITLE
test that first column from augment() is the prediction

### LINKED ISSUE #800: augment predictions not in the same order
Even with #799, the predictions are not in the same order, as `augment.tune_results()` and `augment.model_fit()` disagrees. This is happening because `collect_predictions()` puts `.pred_class` after `.pred_x` and `.pred_y` where `augment.model_fit()` puts it before.

``` r
library(tidymodels)

lr_spec <- parsnip::logistic_reg() %>% 
  parsnip::set_engine("glm")

set.seed(1)
two_class_dat <- as.data.frame(two_class_dat)
bt1 <- rsample::bootstraps(two_class_dat, times = 30)

set.seed(1)
fit_1 <- fit_resamples(
  lr_spec,
  Class ~ .,
  bt1,
  control = control_resamples(save_pred = TRUE)
)

augment(fit_1)
#> # A tibble: 791 × 6
#>    .pred_Class1 .pred_Class2 .pred_class     A     B Class 
#>           <dbl>        <dbl> <fct>       <dbl> <dbl> <fct> 
#>  1        0.509      0.491   Class1       2.07 1.63  Class1
#>  2        0.904      0.0956  Class1       2.02 1.04  Class1
#>  3        0.644      0.356   Class1       1.69 1.37  Class2
#>  4        0.516      0.484   Class1       3.43 1.98  Class2
#>  5        0.425      0.575   Class2       2.88 1.98  Class1
#>  6        0.192      0.808   Class2       3.31 2.41  Class2
#>  7        0.689      0.311   Class1       2.50 1.56  Class2
#>  8        0.547      0.453   Class1       1.98 1.55  Class2
#>  9        0.993      0.00667 Class1       2.88 0.580 Class1
#> 10        0.102      0.898   Class2       3.74 2.74  Class2
#> # ℹ 781 more rows

collect_predictions(fit_1)
#> # A tibble: 8,692 × 7
#>    id          .pred_Class1 .pred_Class2  .row .pred_class Class  .config       
#>    <chr>              <dbl>        <dbl> <int> <fct>       <fct>  <chr>         
#>  1 Bootstrap01       0.892       0.108       2 Class1      Class1 Preprocessor1…
#>  2 Bootstrap01       0.506       0.494       4 Class1      Class2 Preprocessor1…
#>  3 Bootstrap01       0.654       0.346       7 Class1      Class2 Preprocessor1…
#>  4 Bootstrap01       0.991       0.00877     9 Class1      Class1 Preprocessor1…
#>  5 Bootstrap01       0.0732      0.927      10 Class2      Class2 Preprocessor1…
#>  6 Bootstrap01       0.0829      0.917      12 Class2      Class2 Preprocessor1…
#>  7 Bootstrap01       0.851       0.149      14 Class1      Class1 Preprocessor1…
#>  8 Bootstrap01       0.759       0.241      18 Class1      Class1 Preprocessor1…
#>  9 Bootstrap01       0.0672      0.933      21 Class2      Class2 Preprocessor1…
#> 10 Bootstrap01       0.168       0.832      23 Class2      Class2 Preprocessor1…
#> # ℹ 8,682 more rows

lr_spec %>%
  fit(Class ~ ., data = two_class_dat) %>%
  augment(two_class_dat)
#> # A tibble: 791 × 6
#>    .pred_class .pred_Class1 .pred_Class2     A     B Class 
#>    <fct>              <dbl>        <dbl> <dbl> <dbl> <fct> 
#>  1 Class1             0.513      0.487    2.07 1.63  Class1
#>  2 Class1             0.906      0.0940   2.02 1.04  Class1
#>  3 Class1             0.645      0.355    1.69 1.37  Class2
#>  4 Class1             0.599      0.401    3.43 1.98  Class2
#>  5 Class2             0.435      0.565    2.88 1.98  Class1
#>  6 Class2             0.201      0.799    3.31 2.41  Class2
#>  7 Class1             0.701      0.299    2.50 1.56  Class2
#>  8 Class1             0.563      0.437    1.98 1.55  Class2
#>  9 Class1             0.994      0.00622  2.88 0.580 Class1
#> 10 Class2             0.105      0.895    3.74 2.74  Class2
#> # ℹ 781 more rows
```

<sup>Created on 2024-01-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
This PR aims to close #800 by adding the tests needed.

Right now it will fail CI, but once https://github.com/tidymodels/tune/pull/798 has been merged in, it should run cleanly.

### CODE REVIEW

File: R/collect.R [hunk 1]
```diff
@@ -152,7 +152,13 @@ collect_predictions.tune_results <- function(x, summarize = FALSE, parameters =
   } else {
     x <- collector(x, coll_col = coll_col)
   }
-  dplyr::relocate(x, dplyr::starts_with(".pred"))
+
+  x <- dplyr::relocate(x, dplyr::starts_with(".pred"))
```
Comment (topepo) [inline R/collect.R:156]: I played around with how to do this too. I think the order should be

 1. `.pred`
 2. `.pred_class`, `.pred_time` (and so on for other non class-probability estimates)
 3. `.pred_{class level}`
 
I just thought of this, which seems to work: 

``` r
library(tidymodels)
test <- tibble(.pred_a = 1:3, .pred_b = 1:3, .pred_class = 1:3, 
               .pred = 1:3, .pred_time = 1:3)
test %>% select(any_of(".pred"), any_of(".pred_class"), 
                any_of(".pred_time"), starts_with(".pred"))
#> # A tibble: 3 × 5
#>   .pred .pred_class .pred_time .pred_a .pred_b
#>   <int>       <int>      <int>   <int>   <int>
#> 1     1           1          1       1       1
#> 2     2           2          2       2       2
#> 3     3           3          3       3       3
```

<sup>Created on 2024-01-12 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

File: tests/testthat/test-augment.R [hunk 2]
```diff
@@ -26,6 +26,8 @@ test_that("augment fit_resamples", {
   expect_true(sum(names(aug_1) == ".resid") == 0)
   expect_s3_class_bare_tibble(aug_1)
 
+  expect_true(names(aug_1)[1] == ".pred_class")
+
   expect_identical(
     rle(grepl("^\\.", names(aug_1)))$values,
```
Comment (topepo) [inline tests/testthat/test-augment.R:32]: Just out of curiosity, do we know what this is testing?

Comment (EmilHvitfeldt) [reply tests/testthat/test-augment.R:32]: if you are referring to line 32, it test that columns that start with `.` comes first https://github.com/tidymodels/tune/pull/799

File: R/collect.R (hunk 1 above)
Comment (EmilHvitfeldt) [reply R/collect.R]: done in [`4376325` (#803)](https://github.com/tidymodels/tune/pull/803/commits/4376325dbeee428c10be81ef31adfc08ddf699af)

**Pull request merged**